### PR TITLE
Add unit tests for com.amazon.opendistroforelasticsearch.security.support.ModuleInfo

### PR DESCRIPTION
**Pull request by Braavos6**
I've analysed your codebase and noticed that `com.amazon.opendistroforelasticsearch.security.support.ModuleInfo` is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -70,6 +70,7 @@
     
     <!-- Test only -->
     <mockito.version>1.10.19</mockito.version>
+    <powermockito.version>1.6.6</powermockito.version>
```
Comment (petersear) [inline pom.xml]: I think this should be `powermock.version`.

File: pom.xml [hunk 2]
```diff
@@ -223,6 +224,12 @@
       <scope>test</scope>
     </dependency>
 
+    <dependency>
+      <groupId>org.powermock</groupId>
+      <artifactId>powermock-api-mockito</artifactId>
+      <version>${powermockito.version}</version>
```
Comment (petersear) [inline pom.xml]: `powermock.version`

File: src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.security.support;
+
+import com.amazon.opendistroforelasticsearch.security.auth.HTTPAuthenticator;
```
Comment (petersear) [inline src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java:33]: Consider breaking the imports up into groups. Looking at other test files, they seem to leave a blank line between groups, starting with java.*, then org.*, then com.*

File: src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        hashMap.put("actual_implementation", "classname");
+        hashMap.put("version", "");
+        hashMap.put("buildTime", "");
+        hashMap.put("gitsha1", "");
+
+        Assert.assertEquals(hashMap, moduleInfo.getAsMap());
+    }
+
+    @Test
+    public void testGetAsMap2() throws IOException {
```
Comment (petersear) [inline src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java:64]: The source method does not declare that it `throws IOException`, nor does the other test method for this method, testGetAsMap1()

File: src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    @Test
+    public void testHashCode1() {
+        ModuleType moduleType = ModuleType.getByDefaultImplClass(String.class);
+        ModuleInfo moduleInfo = new ModuleInfo(moduleType, "classname");
+
+        Assert.assertEquals(1248676478, moduleInfo.hashCode());
+    }
+
+    @Test
+    public void testHashCode2() throws IOException {
```
Comment (petersear) [inline src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java:94]: Is `throws` necessary?

File: src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+        PowerMockito.when(streamInput.readEnum(ModuleType.class))
+                .thenReturn(moduleType);
+        ModuleInfo moduleInfo = new ModuleInfo(streamInput);
+
+        Assert.assertEquals(-793453215, moduleInfo.hashCode());
+    }
+
+    @Test
+    public void testEquals1() {
+        ModuleType moduleType1 = ModuleType.getByDefaultImplClass(String.class);
```
Comment (petersear) [inline src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java]: I *think* this line is 81 chars long

File: src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+
+        Assert.assertFalse(moduleInfo1.equals(null));
+        Assert.assertFalse(moduleInfo1.equals("foo"));
+        Assert.assertFalse(moduleInfo3.equals(moduleInfo1));
+        Assert.assertFalse(moduleInfo1.equals(moduleInfo3));
+        Assert.assertFalse(moduleInfo1.equals(moduleInfo2));
+    }
+
+    @Test
+    public void testEquals2() throws IOException {
```
Comment (petersear) [inline src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java:127]: Is `throws` needed?

File: src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+        ModuleType moduleType = ModuleType.getByDefaultImplClass(String.class);
+        ModuleInfo moduleInfo = new ModuleInfo(moduleType, "classname");
+
+        Assert.assertEquals(
+                "Module [type=UNKNOWN, implementing class=classname]",
+                moduleInfo.toString());
+    }
+
+    @Test
+    public void testToString2() throws IOException {
```
Comment (petersear) [inline src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java:166]: Is `throws` needed?

**Review by petersear — CHANGES_REQUESTED** [review]

File: src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java (hunk 5 above)
Comment (Braavos6) [reply src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java:94]: StreamInput requires it :(

File: src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java (hunk 6 above)
Comment (Braavos6) [reply src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java]: yeah 81.. changed

File: src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java (hunk 7 above)
Comment (Braavos6) [reply src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java:127]: StreamInput requires it :(

File: src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java (hunk 8 above)
Comment (Braavos6) [reply src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java:166]: StreamInput requires it :(

File: src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java (hunk 4 above)
Comment (Braavos6) [reply src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java:64]: StreamInput requires it :(

File: src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java (hunk 3 above)
Comment (Braavos6) [reply src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java:33]: Done

File: pom.xml (hunk 1 above)
Comment (Braavos6) [reply pom.xml]: I went with `powermockito` as the other one is `mockito` and the `ito` is present :)

Comment (Braavos6) [reply pom.xml]: name changed

File: pom.xml (hunk 2 above)
Comment (Braavos6) [reply pom.xml]: name changed

**Review by petersear — APPROVED** [review]

**Review by JohnLBergqvist — APPROVED** [review]

**Pull request closed without merging**